Can't use typealias in classes annotated with Fixture

Using

```kotlin
    implementation("io.github.bluegroundltd:fixtures-annotations:1.0.8-alpha03")
    ksp("io.github.bluegroundltd:fixtures:1.0.8-alpha03")
```

This works:
```kotlin
@Fixture
@Serializable
data class Foo(
    @Serializable(with = ZonedDateTimeAsStringSerializer::class)
    val dateTime: ZonedDateTime,
)
```

But this (the recommended way https://github.com/Kotlin/kotlinx.serialization/blob/master/docs/serializers.md#specifying-serializer-globally-using-typealias for specifying serializer "globally") doesn't work:
```kotlin
@Fixture
@Serializable
data class Bar(
    val dateTime: ZonedDateTimeAsString,
)
```

the error is:

```
> Task :kspKotlin FAILED
e: [ksp] java.lang.ClassCastException: class com.google.devtools.ksp.symbol.impl.kotlin.KSTypeAliasImpl cannot be cast to class com.google.devtools.ksp.symbol.KSClassDeclaration (com.google.devtools.ksp.symbol.impl.kotlin.KSTypeAliasImpl and com.google.devtools.ksp.symbol.KSClassDeclaration are in unnamed module of loader java.net.URLClassLoader @308df0bb)
        at com.theblueground.fixtures.ProcessedParameterMapper.mapParameter(ProcessedParameterMapper.kt:24)
        at com.theblueground.fixtures.FixtureVisitor.extractParameters(FixtureVisitor.kt:49)
        at com.theblueground.fixtures.FixtureVisitor.visitClassDeclaration(FixtureVisitor.kt:37)
        at com.theblueground.fixtures.FixtureVisitor.visitClassDeclaration(FixtureVisitor.kt:16)
        at com.google.devtools.ksp.symbol.impl.kotlin.KSClassDeclarationImpl.accept(KSClassDeclarationImpl.kt:135)
        at com.theblueground.fixtures.FixtureProcessor.visitFixtures(FixtureProcessor.kt:91)
        at com.theblueground.fixtures.FixtureProcessor.process(FixtureProcessor.kt:59)
        at com.google.devtools.ksp.AbstractKotlinSymbolProcessingExtension$doAnalysis$8$1.invoke(KotlinSymbolProcessingExtension.kt:305)
        at com.google.devtools.ksp.AbstractKotlinSymbolProcessingExtension$doAnalysis$8$1.invoke(KotlinSymbolProcessingExtension.kt:303)
        at com.google.devtools.ksp.AbstractKotlinSymbolProcessingExtension.handleException(KotlinSymbolProcessingExtension.kt:409)
        at com.google.devtools.ksp.AbstractKotlinSymbolProcessingExtension.doAnalysis(KotlinSymbolProcessingExtension.kt:303)
        at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration(TopDownAnalyzerFacadeForJVM.kt:112)
        at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration$default(TopDownAnalyzerFacadeForJVM.kt:77)
        at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1.invoke(KotlinToJVMBytecodeCompiler.kt:256)
        at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1.invoke(KotlinToJVMBytecodeCompiler.kt:247)
        at org.jetbrains.kotlin.cli.common.messages.AnalyzerWithCompilerReport.analyzeAndReport(AnalyzerWithCompilerReport.kt:115)
        at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.analyze(KotlinToJVMBytecodeCompiler.kt:247)
        at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileModules$cli(KotlinToJVMBytecodeCompiler.kt:87)
        at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileModules$cli$default(KotlinToJVMBytecodeCompiler.kt:43)
        at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:165)
        at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:50)
        at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:104)
        at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:48)
        at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:101)
        at org.jetbrains.kotlin.daemon.CompileServiceImpl.compile(CompileServiceImpl.kt:1523)
        at jdk.internal.reflect.GeneratedMethodAccessor88.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at java.rmi/sun.rmi.server.UnicastServerRef.dispatch(UnicastServerRef.java:360)
        at java.rmi/sun.rmi.transport.Transport$1.run(Transport.java:200)
        at java.rmi/sun.rmi.transport.Transport$1.run(Transport.java:197)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:712)
        at java.rmi/sun.rmi.transport.Transport.serviceCall(Transport.java:196)
        at java.rmi/sun.rmi.transport.tcp.TCPTransport.handleMessages(TCPTransport.java:587)
        at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(TCPTransport.java:828)
        at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.lambda$run$0(TCPTransport.java:705)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
        at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(TCPTransport.java:704)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)

e: [ksp] Failed to process fixtures
e: Error occurred in KSP, check log for detail

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':kspKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction
   > Compilation error. See log for more details
```

For completness, here's the serializer:

```kotlin
import com.theblueground.fixtures.Fixture
import kotlinx.serialization.KSerializer
import kotlinx.serialization.Serializable
import kotlinx.serialization.descriptors.PrimitiveKind
import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
import kotlinx.serialization.descriptors.SerialDescriptor
import kotlinx.serialization.encoding.Decoder
import kotlinx.serialization.encoding.Encoder
import java.time.ZonedDateTime
import java.time.format.DateTimeFormatter
import java.time.format.DateTimeFormatter.ISO_LOCAL_DATE
import java.time.format.DateTimeFormatter.ISO_LOCAL_TIME
import java.time.format.DateTimeFormatterBuilder

class ZonedDateTimeAsStringSerializer : KSerializer<ZonedDateTime> {

    private val formatter: DateTimeFormatter = DateTimeFormatterBuilder()
        .parseCaseInsensitive()
        .append(ISO_LOCAL_DATE)
        .appendLiteral(' ')
        .append(ISO_LOCAL_TIME)
        .appendLiteral(" (")
        .appendZoneId()
        .appendLiteral(')')
        .toFormatter()

    override val descriptor: SerialDescriptor =
        PrimitiveSerialDescriptor("ZonedDateTimeAsStringSerializer", PrimitiveKind.STRING)

    override fun serialize(encoder: Encoder, value: ZonedDateTime) {
        encoder.encodeString(value.format(formatter))
    }

    override fun deserialize(decoder: Decoder): ZonedDateTime {
        return ZonedDateTime.parse(decoder.decodeString(), formatter)
    }
}

typealias ZonedDateTimeAsString = @Serializable(with = ZonedDateTimeAsStringSerializer::class) ZonedDateTime
```

--- a/fixtures-processor/src/main/kotlin/com/theblueground/fixtures/ProcessedParameterMapper.kt
+++ b/fixtures-processor/src/main/kotlin/com/theblueground/fixtures/ProcessedParameterMapper.kt
@@ -3,6 +3,7 @@ package com.theblueground.fixtures
 import com.google.devtools.ksp.symbol.KSClassDeclaration
 import com.google.devtools.ksp.symbol.KSDeclaration
 import com.google.devtools.ksp.symbol.KSType
+import com.google.devtools.ksp.symbol.KSTypeAlias
 import com.google.devtools.ksp.symbol.KSValueParameter
 import com.squareup.kotlinpoet.ParameterizedTypeName.Companion.parameterizedBy
 import com.squareup.kotlinpoet.TypeName
@@ -20,19 +21,18 @@ internal class ProcessedParameterMapper(
 
     fun mapParameter(parameterValue: KSValueParameter): ProcessedFixtureParameter {
         val resolvedType = parameterValue.type.resolve()
-        return mapParameter(
-            parameterValue = parameterValue,
-            parameterType = resolvedType,
-            parameterClassDeclaration = (resolvedType.declaration as KSClassDeclaration),
-        )
+        return when (val declaration = resolvedType.declaration) {
+            is KSTypeAlias -> mapParameter(parameterValue, declaration.type.resolve())
+            else -> mapParameter(parameterValue, resolvedType)
+        }
     }
 
     private fun mapParameter(
         parameterValue: KSValueParameter,
         parameterType: KSType,
-        parameterClassDeclaration: KSClassDeclaration,
     ): ProcessedFixtureParameter {
         val name = parameterValue.name!!.asString()
+        val parameterClassDeclaration = parameterType.declaration as KSClassDeclaration
 
         return when {
             parameterType.hasFixtureAdapter(processedFixtureAdapters) -> mapFixtureAdapterParameter(

--- a/fixtures-processor/src/test/kotlin/com/theblueground/fixtures/FixtureProcessorTest.kt
+++ b/fixtures-processor/src/test/kotlin/com/theblueground/fixtures/FixtureProcessorTest.kt
@@ -272,6 +272,51 @@ class FixtureProcessorTest : KSPTest() {
     }
 
     @Test
+    fun `should generate a builder function with resolved type for typealias`() {
+        // Given
+        val fixtureSource = """
+                    package $packageName
+
+                    import com.theblueground.fixtures.Fixture
+
+                    import java.math.BigDecimal
+
+                    @Fixture
+                    data class $fixtureName(
+                        val bigDecimalAliasValue: BigDecimalAlias,
+                    )
+
+                    typealias BigDecimalAlias = BigDecimal
+        """.trimIndent()
+        val fixtureFile = kotlin(name = "$fixtureName.kt", contents = fixtureSource)
+
+        // When
+        val result = compile(
+            arguments = mapOf("fixtures.run" to "true"),
+            sourceFiles = listOf(fixtureFile),
+        )
+        val generatedContent = getGeneratedContent(
+            packageName = packageName,
+            filename = "${fixtureName}Fixture.kt",
+        )
+
+        // Then
+        assertThat(result.exitCode).isEqualTo(KotlinCompilation.ExitCode.OK)
+        val expected = """
+            package $packageName
+
+            import java.math.BigDecimal
+
+            public fun create$fixtureName(bigDecimalAliasValue: BigDecimal = BigDecimal.ZERO): TestClass =
+                $packageName.$fixtureName(
+            	bigDecimalAliasValue = bigDecimalAliasValue
+            )
+
+        """.trimIndent()
+        assertThat(generatedContent).isEqualTo(expected)
+    }
+
+    @Test
     fun `should not generate a builder function while not running tests`() {
         // Given
         val fixtureFile = kotlin(name = "$fixtureName.kt", contents = fixtureSource)

